extensions/PhabBugz/lib/WebService.pm

```diff
@@ -89,19 +91,12 @@ sub revision {
     }
     # else bug is private
     else {
-        my $phab_sync_groups = Bugzilla->params->{phabricator_sync_groups}
-            || ThrowUserError('invalid_phabricator_sync_groups');
-        my $sync_group_names = [ split('[,\s]+', $phab_sync_groups) ];
-
-        my $bug_groups = $bug->groups_in;
-        my $bug_group_names = [ map { $_->name } @$bug_groups ];
-
-        my @set_groups = intersect($bug_group_names, $sync_group_names);
+        my @set_groups = get_security_sync_groups($bug);
 
         # If bug privacy groups do not have any matching synchronized groups,
         # then leave revision private and it will have be dealt with manually.
         if (!@set_groups) {
-            ThrowUserError('invalid_phabricator_sync_groups');
+            add_security_sync_comments(\@revisions, $bug);
         }
 
         my $policy_phid = create_private_revision_policy($bug, \@set_groups);
```

Comment:
So we are getting a failure here in that create_private_revision_policy looks up the project_ids for the bug group set. In this case, the list is empty and so we ThrowUserError(invalid_phabricator_sync_groups). This causes Dorrkeeper to requeue and we get a lot of looping comments.

For this I think we either need to 1) not set a policy at all leaving the current policy in place (most likely 'Admins only') and return or 2) set a custom policy to 'Admins Only' and subscribers here instead. We could do 2) to be sure I suppose in case it had gotten changed in Phab by a user.

In the push connector you use make_revision_private() instead of create_private_revision_policy so we need to match.

Reply:
1.  Could we simply change the code to:

```
# Initialize this result to admin
my $result = 'admin';

# .....

# If bug privacy groups do not have any matching synchronized groups,
# then leave revision private and it will have be dealt with manually.
if (!@set_groups) {
    add_security_sync_comments(\@revisions, $bug);
}
else {
    my $policy_phid = create_private_revision_policy($bug, \@set_groups);
    my $subscribers = get_bug_role_phids($bug);
    $result = edit_revision_policy($revision_phid, $policy_phid, $subscribers);
}
```

...and then let it keep going?

2.

Reply:
1. I thought about that and normally we could just do nothing. But we could also just set the policy to admin + subscribers. It would still be a custom policy. This would allow the assignee, qa contact, and cc members to see the revision even though it doesnt have any phab project (groups) associated with it. If we do nothing then the subscribers are not added and they would not be able to see the revision anyway.

2. Yeah maybe it doesnt matter here. I was saying because it would be a shortcut to making the revision private to admin/admin if @set_groups is empty. We use that in Push connector when a bugs groups are changed to a set of group which are not being synced. Which sounded very similar to what is occurring in WebService.pm as well.